### PR TITLE
Added file extension detection through filename parameter

### PR DESCRIPTION
**Pull request by okrad**
I'm working on a project in wich the exam url is not referencing directly the file on the server, but points to a page that determines the proper exam to return through a query string parameter, for example: `https://www.myserver.com/getExam?id=1`
This wa, ami can't detect the file type by examinig the file extension.

With this PR I added a patch for detecting the extension using a standard query string parameter "filename", so I can slightly change my urls like this: `https://www.myserver.com/getExam?id=1&filename=exam1.dicom`

### CODE REVIEW

**Comment by NicolasRannou** [comment]
Thanks for looking into it!

Instead of adding 'filename=exam1.dicom' to the URL, can you add "contentType=application%2Fdicom"? I believe that would solve it: https://github.com/okrad/ami/blob/c1350caefac2154f427a07204ee9c96e3b01e9fd/src/core/core.utils.js#L149-L153

**Comment by NicolasRannou** [comment]
`https://www.myserver.com/getExam?id=1&contentType=application/dicom`

**Comment by okrad** [comment]
Yes, I thought about using the contentType parameter. 
I must admit my PR was a little misleading, since I'm not using dicom format. Actually my exams are nifti, for whom it doesn't seem to exist a standard mime type...

**Comment by NicolasRannou** [comment]
OK, any idea why the tests are failing now?

File: src/core/core.utils.js [hunk 1]
```diff
@@ -115,11 +116,23 @@ export default class CoreUtils {
     data.query = '';
 
     let parsedUrl = URL.parse(url);
+
     data.pathname = parsedUrl.pathname;
     data.query = parsedUrl.query;
 
+    //Find "filename" parameter value, if present
+    data.filename = data.query.split('&').reduce((acc, fieldval) => {
```
Comment (NicolasRannou) [inline src/core/core.utils.js]: we have to make sure `data.query` is not `undefined`

**Comment by NicolasRannou** [comment]
Last things, maybe also add as a comment which type of URL it is parsing!

Thanks!

**Pull request merged**